### ISSUE TITLE
pytorch video dataset + pose and data loader equivalent 

### ISSUE THREAD

**Issue by mikatsukons** [NVIDIA/DALI#4761]
hello. 
I have a dataset which reads video paths, action_IDs from a .csv file and saves them to a dictionary (named data)  and in get_item I want to do the following steps:
1. get a video path (indexing the keys of data dictionary)
2. get the corresponding action_ID from the dictionary data
3. load the video frames in 4D torch.tensors [frames, channels, width, height]
4. do some other stuff (e.g., sampling the frames indices - but not selecting the frames)
5. transform data (random affine, colour jitter, etc etc)
6. **load skeleton pickle file (using the video path -- ofc modified e.g data/dataset/train/video1.avi -->data/dataset/poses/train/video1_mediapipe.pkl)**
7. **augment skeleton (normalise, random affine etc etc)**
8. return (video, skeleton), actionID, segment_indices etc etc

So the thing is I want to load a video from a video path and use the same video path to do some other stuff load other files (on condition), transform the video and the other loaded data etc.

I cannot find any similar works to see how it should be done. Is there any available template that manipulates data in equivalent get_item?
Thanks

**Comment by awolant** [comment]
Hello, 
thanks for asking the question. 

You could try to combine [external source operator](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/general/data_loading/external_input.html?highlight=external%20source) and [video decoder operator](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/operations/nvidia.dali.fn.experimental.inputs.video.html?highlight=video%20input#nvidia.dali.fn.experimental.inputs.video) to manually load videos. 
And use external source again to load unpickled files.
Then you can follow [this tutorial](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/math/geometric_transforms.html) to learn how to do geometric transforms on both frames and the skeleton to keep them consistent.
You need to [wrap your pipeline ](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/frameworks/pytorch/pytorch-basic_example.html) to get PyTorch tensors at the output.

Hope that will get you started.
